k9mail-library/src/main/java/com/fsck/k9/mail/store/imap/ImapFolder.java

```diff
@@ -839,8 +840,13 @@ public void fetchPart(Message message, Part part, MessageRetrievalListener<Messa
                             String contentTransferEncoding =
                                     part.getHeader(MimeHeader.HEADER_CONTENT_TRANSFER_ENCODING)[0];
                             String contentType = part.getHeader(MimeHeader.HEADER_CONTENT_TYPE)[0];
-                            MimeMessageHelper.setBody(part, MimeUtility.createBody(bodyStream, contentTransferEncoding,
-                                    contentType));
+                            if(progressCallback != null) {
```

Comment:
This check isn't necessary - if it's null you're passing null, if it's not null you're passing the value.